### PR TITLE
DOC: Add examples for ``np.polynomial.polynomial`` functions

### PR DESCRIPTION
**Pull request by linus-md**
This adds examples for several functions in ``np.polynomial.polynomial``.

### CODE REVIEW

File: numpy/polynomial/polynomial.py [hunk 1]
```diff
@@ -312,6 +312,12 @@ def polymulx(c):
 
     .. versionadded:: 1.5.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial import polynomial as P
+    >>> c = (1,2,3)
```
Comment (eendebakpt) [inline numpy/polynomial/polynomial.py:318]:
```suggestion
    >>> c = (1, 2, 3)
```
autopep8 (similar at several other places)

File: numpy/polynomial/polynomial.py [hunk 2]
```diff
@@ -891,6 +897,12 @@ def polyval2d(x, y, c):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial import polynomial as P
+    >>> c = ((1,2,3),(4,5,6))
```
Comment (eendebakpt) [inline numpy/polynomial/polynomial.py:903]:
```suggestion
    >>> c = ((1, 2, 3), (4, 5, 6))
```

File: numpy/polynomial/polynomial.py [hunk 3]
```diff
@@ -944,6 +956,13 @@ def polygrid2d(x, y, c):
...
```
Comment (eendebakpt) [inline numpy/polynomial/polynomial.py:962]:
```suggestion
    >>> c = ((1, 2, 3), (4, 5, 6))
```

File: numpy/polynomial/polynomial.py [hunk 4]
```diff
@@ -1051,6 +1077,13 @@ def polygrid3d(x, y, z, c):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial import polynomial as P
+    >>> c = ((1,2,3),(4,5,6),(7,8,9))
```
Comment (eendebakpt) [inline numpy/polynomial/polynomial.py:1083]:
```suggestion
    >>> c = ((1, 2, 3), (4,5,6), (7, 8, 9))
```

File: numpy/polynomial/polynomial.py [hunk 5]
```diff
@@ -995,6 +1014,13 @@ def polyval3d(x, y, z, c):
...
```
Comment (eendebakpt) [inline numpy/polynomial/polynomial.py:1020]:
```suggestion
    >>> c = ((1, 2, 3), (4, 5, 6), (7, 8, 9))
```

File: numpy/polynomial/polynomial.py [hunk 6]
```diff
@@ -1051,6 +1077,13 @@ def polygrid3d(x, y, z, c):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial import polynomial as P
+    >>> c = ((1, 2, 3), (4,5,6), (7, 8, 9))
```
Comment (eendebakpt) [inline numpy/polynomial/polynomial.py:1083]:
```suggestion
    >>> c = ((1, 2, 3), (4, 5, 6), (7, 8, 9))
```

**Review by eendebakpt — APPROVED** [review]

**Review by mtsokol — APPROVED** [review]

**Comment by mtsokol** [comment]
Last thing - please squash PR into one commit with `DOC: ...` message.

**Review by rgommers — APPROVED** [review]
LGTM too. Thanks for the PR @linus-md and for the reviews @eendebakpt and @mtsokol.

**Pull request merged**